calrissian/k8s.py

```diff
@@ -89,6 +92,7 @@ def should_delete_pod(self):
         else:
             return True
 
+    @retry_exponential_if_exception_type((ApiException, HTTPError,), log)
     def delete_pod_name(self, pod_name):
         try:
             self.core_api_instance.delete_namespaced_pod(pod_name, self.namespace)
```

Comment:
We are retrying delete for all ApiException/HTTPErrors. For cases where the pod is not found(404) we do not need to retry.